collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_apiclient.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2020, Simon Dodsley (simon@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: purefa_apiclient
+version_added: '1.5.0'
+short_description: Manage FlashArray API Clients
+description:
+- Enable or disable FlashArray SMI-S Provider and/or SLP
+author:
+- Pure Storage Ansible Team (@sdodsley) <pure-ansible-team@purestorage.com>
+options:
+  name:
+    description:
+    - Name of the API Client
+    type: str
+    required: true
+  state:
+    description:
+    - Define whether the volume should exist or not.
+    default: present
+    choices: [ absent, present ]
+    type: str
+  role:
+    description:
+    - The maximum role allowed for ID Tokens issued by this API client
+    type: str
+    choices: [readonly, ops_admin, storage_admin, array_admin]
+  issuer:
```

Comment:
The role is only accepted during creation and ignored if the API client already exists. Is that intentional? Can also not be changed using the GUI for an existing API client btw. However I would expect an error or warning?

Reply:
Yes - you cannot change an API client once it is created. All you can do is enable/disable the client